### PR TITLE
Upgrade client core versions

### LINKED ISSUE #56: Upgrade the angular-cli and server npm packages (6)
This issue should fix the client build in #46 
Upgrade to angular 5.

### PR DESCRIPTION
**Pull request by DavidBronfen**
Fix #56

### CODE REVIEW

**Comment by DavidBronfen** [comment]
@nirgn975
All the system has been upgraded.
The test fails because of the ngrx spec files that still need to be updated.

**Comment by nirgn975** [comment]
@DavidBronfen So update them or delete them. Your choice.

**Comment by nirgn975** [comment]
build__311_-_davidbronfen_retrogusto_-_travis_ci

File: client/src/app/modules/recipes/services/recipes.service.spec.ts [hunk 1]
```diff
@@ -1,30 +1,49 @@
-import { TestBed, async, inject } from '@angular/core/testing';
-import { HttpClientModule, HttpClient } from '@angular/common/http';
+import { TestBed, getTestBed } from '@angular/core/testing';
```
Comment (nirgn975) [inline client/src/app/modules/recipes/services/recipes.service.spec.ts]: I'm so proud that you didn't go the `inject` way, I'm about to cry.

File: client/src/app/modules/recipes/services/recipes.service.spec.ts [hunk 2]
```diff
@@ -1,30 +1,49 @@
...
+      "rating":4 ,
+      "description":"ריבולטה הוא המרק הטוסקני המפורסם ביותר שהלכה למעשה מדובר בנזיד המורכב מירקות ולחם. השפית לירז שדה (מבראסרי עין כרם) מציעה את הגרסה הירושלמית",
+      "prep_time":"2.5",
+      "portions":8,
+      "calories":460
+    }];
+
+    service.getRecipes().subscribe(recipe => {
+      expect(recipe).toBe(dummyRecipe);
+
```
Comment (nirgn975) [inline client/src/app/modules/recipes/services/recipes.service.spec.ts]: Why you have a whitespace over here?

File: client/src/app/services/categories.service.spec.ts [hunk 3]
```diff
@@ -1,30 +1,45 @@
...
+    const dummyCategory = [{
+      "id":1,
+      "name_he":"ארוחת בוקר",
+      "name_en":"breakfast",
+      "image_path":"data/categories/breakfast.png"
+    }];
+
+    service.getCategories().subscribe(categories => {
+      expect(categories).toBe(dummyCategory);
+
```
Comment (nirgn975) [inline client/src/app/services/categories.service.spec.ts]: Why you have a whitespace here?

File: client/src/app/effects/categories.ts [hunk 4]
```diff
@@ -1,8 +1,11 @@
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/switchMap';
 import { Injectable } from '@angular/core';
-import { Effect, Actions } from '@ngrx/effects';
+import { Effect, Actions, toPayload } from '@ngrx/effects';
```
Comment (nirgn975) [inline client/src/app/effects/categories.ts]: why you have `toPayload` here? It's [deprecated](https://github.com/ngrx/platform/blob/master/docs/effects/api.md#topayload-deprecated)

File: client/src/app/modules/recipes/reducers/index.ts [hunk 5]
```diff
@@ -47,43 +23,104 @@ export interface State {
...
- * 	  this.booksState$ = state$.select(getBooksState);
- * 	}
- * }
- * ```
+ * Layout Reducers
  */
-export const getRecipesState = (state: State) => state.recipes;
+export const getRecipesState = createFeatureSelector<fromRecipes.State>('recipes');
+
+// => => => => => => => => => => => => => => => => => => => => => => => => => => => =>
```
Comment (nirgn975) [inline client/src/app/modules/recipes/reducers/index.ts]: WHY? "in your mama", WHY?

Comment (DavidBronfen) [reply client/src/app/modules/recipes/reducers/index.ts]: hahahahah!!

File: client/src/app/modules/recipes/services/recipes.service.spec.ts (hunk 1 above)
Comment (DavidBronfen) [reply client/src/app/modules/recipes/services/recipes.service.spec.ts]: Why are you mistalbet?

**Pull request merged**